Replace deprecated String.prototype.substr()

CommanderRoot opened the pull request:
[String.prototype.substr()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) is deprecated so we replace it with [String.prototype.slice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) which works similarily but isn't deprecated.
.substr() probably isn't going away anytime soon but the change is trivial so it doesn't hurt to do it.

CommanderRoot committed:
Replace deprecated String.prototype.substr()

.substr() is deprecated so we replace it with .slice() which works similarily but isn't deprecated

Signed-off-by: Tobias Speicher <rootcommander@gmail.com>

nightpool commented:
is there an eslint rule or similar we can use to avoid running into these kind of issues in the future?

CommanderRoot committed:
Change String.prototype.substring() to String.prototype.slice()

.substring() and .slice() work very similary but .slice() is a bit faster and stricter

Signed-off-by: Tobias Speicher <rootcommander@gmail.com>

CommanderRoot committed:
Add ESLint rule to forbid usage of .substr and .substring

.substr() is deprecated and .substring() is very similar to .slice() so better to use .slice() at all times

Signed-off-by: Tobias Speicher <rootcommander@gmail.com>

CommanderRoot commented:
Yes there is. I have added it to the PR. The rules are from [here](https://github.com/eslint/eslint/blob/8bb527660ffddc122204fb5bf59076687aaa78b8/packages/eslint-config-eslint/default.yml#L187-L188)

nightpool commented:
@CommanderRoot just out of curiosity, is there a reason you're going around and opening these PRs on thousands of Javascript projects? Just because you can?

CommanderRoot commented:
I noticed the deprecation of the `substr()` function in my IDE so I started to update my own projects. After that I "experimented" with updating some of the other projects and tools I use.
Currently I just do it when I'm bored as it's fairly relaxing and also not wasted time.

nightpool commented:
Thanks again!
